src/app/api/comments/route.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable prettier/prettier */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import prisma from "@/lib/prisma";
+import { NextResponse, type NextRequest } from "next/server";
+import { createCommentSchema } from "./schema";
+
+//to create a comment
+export async function POST(req: NextRequest) {
+  try {
+    console.log("Received request:", req);
+    const requestBody = await req.json();
+    console.log("Request body:", requestBody);
+
+    const parsed = createCommentSchema.safeParse(requestBody);
+    console.log("Validation result:", parsed);
+
+    if (!parsed.success) {
+      console.error("Validation error:", parsed.error);
+      return new NextResponse(JSON.stringify({ error: parsed.error }), {
+        status: 400,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+    }
+
+    const commentData = parsed.data;
+
+    const createdComment = await prisma.comment.create({
+      data: {
+        content: commentData.content,
+        quote: commentData.quote,
+        parentCommentId: commentData.parentCommentId,
+        userId: commentData.userId,
```

Comment:
I put a bunch of these comments, so I'm sorry for being repetitive, just not sure if you'll see all these instances, but it may be good to check current auth session and also use user email in the main request because the front end only has email and names I think